src/Test/test_conic.jl

```diff
@@ -6831,3 +6831,165 @@ function setup_test(
     )
     return
 end
+
+"""
+    test_conic_HermitianPositiveSemidefiniteConeTriangle_1(model::MOI.ModelLike, config::Config)
+
+Test the computation of the projection of a hermitian matrix to the cone
+of hermitian positive semidefinite matrices.
+The matrix is:
+```
+[ 1    -1+im]   [ 1-im  ?]   [√3    ]   [ 1+im  -1+√3]
+[-1-im -1   ] = [-1+√3  ?] * [   -√3] * [  ?      ?  ] / (6 - 2√3)
+```
+so it's projection to the Hermitian PSD cone is:
+```
+                 [1-im]  [ 1+im  1-√3]
+√3 / (6 - 2√3) * [1-√3]
+```
+which is:
+```
+                 [1-im]  [ 1+im  1-√3]
+(1 + √3) / 4   * [1-√3]
+```
+which is
+```
+[(1+√3)/2  -1/2+im/2]
+[-1/2+im/2 (-1+√3)/2]
+```
+"""
+function test_conic_HermitianPositiveSemidefiniteConeTriangle_1(optimizer::MOI.ModelLike, config::Config{T}) where {T}
+    atol = config.atol
+    rtol = config.rtol
+
+    MOI.empty!(optimizer)
+    set = MOI.HermitianPositiveSemidefiniteConeTriangle(2)
+    x, cx = MOI.add_constrained_variables(optimizer, set)
+    x11 = x[1:3]
+    x12 = x[4]
+    t = MOI.add_variable(optimizer)
+    soc_set = MOI.SecondOrderCone(5)
+    con_soc = MOI.add_constraint(
+        optimizer,
+        MOI.Utilities.operate(
+            vcat,
+            T,
+            t,
+            x[1] - one(T),
+            √T(2) * (x[2] + one(T)),
+            x[3] + one(T),
+            √T(2) * (x[4] - one(T)),
+        ),
+        soc_set,
+    )
+    MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.set(optimizer, MOI.ObjectiveFunction{typeof(t)}(), t)
+    if _supports(config, MOI.optimize!)
+        MOI.optimize!(optimizer)
+        primal = [(T(1) + √T(3)) / T(2), -T(1) / T(2), (-T(1) + √T(3)) / T(2), T(1) / T(2)]
+        soc_primal = [
+            primal[1] - one(T),
+            √T(2) * (primal[2] + one(T)),
+            primal[3] + one(T),
+            √T(2) * (primal[4] - one(T)),
+        ]
+        t_value = LinearAlgebra.norm(soc_primal)
+        soc_primal = [t_value; soc_primal]
+        dual = [(T(3) - √T(3)) / T(6), √T(3)/T(6), (T(3) + √T(3)) / T(6), -√T(3)/T(6)]
+        @test MOI.Utilities.set_dot(primal, dual, set) ≈ 0 atol = atol rtol = rtol
+        soc_dual = [one(T), -dual[1], -dual[2]*√T(2), -dual[3], -dual[4]*√T(2)]
+        @test MOI.Utilities.set_dot(soc_primal, soc_dual, soc_set) ≈ 0 atol = atol rtol = rtol
+        @test MOI.get(optimizer, MOI.VariablePrimal(), x) ≈ primal atol = atol rtol =
```

Comment:
```suggestion
        @test ≈(MOI.get(optimizer, MOI.VariablePrimal(), x), primal, config)
```

I added a method to `config` to simplify these sorts of lines.